ext/opcache/ZendAccelerator.c

```diff
@@ -3635,6 +3489,120 @@ static void preload_remove_empty_includes(void)
 	} ZEND_HASH_FOREACH_END();
 }
 
+static int preload_optimize(zend_persistent_script *script)
+{
+	if (!zend_optimize_script(&script->script, ZCG(accel_directives).optimization_level, ZCG(accel_directives).opt_debug_level)) {
+		return FAILURE;
+	}
+	ZEND_HASH_FOREACH_PTR(preload_scripts, script) {
+		if (!zend_optimize_script(&script->script, ZCG(accel_directives).optimization_level, ZCG(accel_directives).opt_debug_level)) {
+			return FAILURE;
+		}
+	} ZEND_HASH_FOREACH_END();
+	return SUCCESS;
+}
+
+static zend_persistent_script* preload_script_in_shared_memory(zend_persistent_script *new_persistent_script)
+{
+	zend_accel_hash_entry *bucket;
+	uint32_t memory_used;
+	uint32_t checkpoint;
+
+	/* exclusive lock */
+	zend_shared_alloc_lock();
+
+	if (zend_accel_hash_is_full(&ZCSG(hash))) {
+		zend_shared_alloc_unlock();
+		zend_accel_error(ACCEL_LOG_FATAL, "No enogh entries in hash table for preloading!");
+		return NULL;
+	}
+
+	checkpoint = zend_shared_alloc_checkpoint_xlat_table();
+
+	/* Calculate the required memory size */
+	memory_used = zend_accel_script_persist_calc(new_persistent_script, NULL, 0, 1);
+
+	/* Allocate shared memory */
+#if defined(__AVX__) || defined(__SSE2__)
+	/* Align to 64-byte boundary */
+	ZCG(mem) = zend_shared_alloc(memory_used + 64);
+	if (ZCG(mem)) {
+		ZCG(mem) = (void*)(((zend_uintptr_t)ZCG(mem) + 63L) & ~63L);
+#if defined(__x86_64__)
+		memset(ZCG(mem), 0, memory_used);
+#elif defined(__AVX__)
+		{
+			char *p = (char*)ZCG(mem);
+			char *end = p + memory_used;
+			__m256i ymm0 = _mm256_setzero_si256();
+
+			while (p < end) {
+				_mm256_store_si256((__m256i*)p, ymm0);
+				_mm256_store_si256((__m256i*)(p+32), ymm0);
+				p += 64;
+			}
+		}
+#else
+		{
+			char *p = (char*)ZCG(mem);
+			char *end = p + memory_used;
+			__m128i xmm0 = _mm_setzero_si128();
+
+			while (p < end) {
+				_mm_store_si128((__m128i*)p, xmm0);
+				_mm_store_si128((__m128i*)(p+16), xmm0);
+				_mm_store_si128((__m128i*)(p+32), xmm0);
+				_mm_store_si128((__m128i*)(p+48), xmm0);
+				p += 64;
+			}
+		}
+#endif
+	}
+#else
+	ZCG(mem) = zend_shared_alloc(memory_used);
+	if (ZCG(mem)) {
+		memset(ZCG(mem), 0, memory_used);
+	}
+#endif
+	if (!ZCG(mem)) {
+		zend_shared_alloc_unlock();
+		zend_accel_error(ACCEL_LOG_FATAL, "No enogh shared memory for preloading!");
```

Comment:
No enough shared memory for preloading

Reply:
@raziel057 No > Not

Reply:
Then maybe
*Not enough shared memory for preloading*

:grin: